src/data/fintype/basic.lean

```diff
@@ -1016,3 +1016,37 @@ instance nat.infinite : infinite ℕ :=
 
 instance int.infinite : infinite ℤ :=
 infinite.of_injective int.of_nat (λ _ _, int.of_nat.inj)
+
+section trunc
+
+/--
+For `s : multiset α`, we can lift the existential statement that `∃ x, x ∈ s` to a `trunc α`.
+-/
+def trunc_of_multiset_exists_mem {α} (s : multiset α) : (∃ x, x ∈ s) → trunc α :=
+quotient.rec_on_subsingleton s $ λ l h,
+  match l, h with
+    | [],       _ := false.elim (by tauto)
+    | (a :: _), _ := trunc.mk a
+  end
+
+/--
+A `nonempty` `fintype` constructively contains an element.
+-/
+def trunc_of_nonempty_fintype {α} (h : nonempty α) [fintype α] : trunc α :=
```

Comment:
Yes, because of `trunc_of_card_pos` immediately below. Also, some argument should be explicit here: the other alternative would be `(α)`. I'm inclined to leave as is.

Reply:
I think I would actually rather have `(\a)`, and then you could either use `@` below, or a `letI : nonempty \a := bla; exact _`

Reply:
@jcommelin, could you check https://github.com/leanprover-community/mathlib/pull/3166/commits/3694bb32d9541586993ffb4e285081a4f40ff78c and see if you think it's an improvement?